src/libpmemobj/sync.c

```diff
@@ -632,13 +650,17 @@ pmemobj_cond_wait(PMEMobjpool *pop, PMEMcond *condp,
 	PMEMmutex_internal *mutexip = (PMEMmutex_internal *)mutexp;
 	os_cond_t *cond = get_cond(pop, condip);
 	os_mutex_t *mutex = get_mutex(pop, mutexip);
-	if ((cond == NULL) || (mutex == NULL))
+	if ((cond == NULL) || (mutex == NULL)) {
+		PMEMOBJ_API_END();
 		return EINVAL;
+	}
 
 	ASSERTeq((uintptr_t)mutex % util_alignof(os_mutex_t), 0);
 	ASSERTeq((uintptr_t)cond % util_alignof(os_cond_t), 0);
 
-	return os_cond_wait(cond, mutex);
+	int ret = os_cond_wait(cond, mutex);
```

Comment:
cond_wait (and some other APIs) should not modify pmem, so there's no point in adding markers for them